### ISSUE TITLE
Mark all as read/do not count unread on login

### ISSUE THREAD

**Issue by AdminRAT** [Gruntfuggly/discord-chat#6]
Sometimes it can take a long time to load all of the unread messages (count) if you have a lot in some servers. It would be nice if there was an option to mark all messages as read/not count unread messages on login/startup.

**Comment by Gruntfuggly** [comment]
Hi - I've got a chance to look at this. I could add an option to not bother loading the unread messages. Then all channels will appear to be read at startup. New messages arriving after startup would be the only messages considered as unread. Does that sound useful?

**Comment by Gruntfuggly** [comment]
I've pushed an update with a new setting `discord-chat.fetchUnreadMessages` which you can set to false to disable loading the unread message counts.

Let me know if that works for you.

**Comment by AdminRAT** [comment]
I tried it and it seems to work out a lot faster. I did notice that when I did click on a channel to "view" it, it does not mark the channel as read. Is that normal, or do I have to right click on the channel every time and mark it as read, or something else I'm missing/not noticing?

**Comment by Gruntfuggly** [comment]
Yes, originally it would mark the channel as read when you opened it, but I found it was a bit too aggressive. At the moment you need to either use the tick icon to mark the channel read, or you can mark all the channels on the server as read. You can also right click and mark them read from a context menu.

If you can think of a better compromise, I'd be interested. I was thinking of trying to put a context menu in the output view itself maybe?

**Comment by AdminRAT** [comment]
That might seem like an idea. I also was thinking of opening the chat in a separate tab, although I'm not sure if it's possible to have the chat and message bar in it (somewhat like the output view but with a message bar)

**Comment by Gruntfuggly** [comment]
Using the output window for the chats makes things quite simple as you can just append new entries really easily - using a new tab would be quite a big change and a whole lot more work I think.

What do you mean by message bar? the side bar?

**Comment by AdminRAT** [comment]
What I meant by the message bar is this:
![image](https://user-images.githubusercontent.com/24538037/58134770-ac955480-7bdc-11e9-9ca1-7248ecb958a1.png)

**Comment by Gruntfuggly** [comment]
Oh, OK. Definitely a bit more work! :grin:

**Comment by Gruntfuggly** [comment]
I've just uploaded a version with some additions to the  context menu for the output channels.

**Comment by Gruntfuggly** [comment]
I've just uploaded a new version which improves the performance a bit - it no longer loads the unread messages for muted channels.

I'm going to close this issue now as the main point was fixed - having the message bar and chat in a separate tab would be a major feature to implement, so I don't propose to do it in the near future (too many projects on the go!). Feel free to raise it as a separate issue though - that way if other people want it too, it can get +1d.
